### PR TITLE
Add Dorgesh-Kaan agility course overlay

### LINKED ISSUE #3440: Add Dorgesh-Kaan Agility course overlay
It would be a nice feature to have an overlay for the course that shows which part the engineer wants you to retrieve. It should be fairly straightforward to implement.

### PR DESCRIPTION
**Pull request by Hyftar**
This pull request adds a simple overlay in the upper left corner, when the player is within the Dorgesh-Kaan agility course region, that displays the last part the engineer required the player to retrieve.

screenshot from 2018-05-29 22-32-44

This PR closes #3440

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/agility/DorgeshKaanOverlay.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.agility;
+
+import java.awt.*;
```
Comment (LeviSchuck) [inline runelite-client/src/main/java/net/runelite/client/plugins/agility/DorgeshKaanOverlay.java]: Need to adjust IntelliJ settings or whatever to not multi import

Comment (Hyftar) [reply runelite-client/src/main/java/net/runelite/client/plugins/agility/DorgeshKaanOverlay.java]: Oops missed that one.

**Comment by Hyftar** [comment]
It is within the original agility code. :)

**Comment by LeviSchuck** [comment]
My bad. 
Apart from the style issue, I support this PR.

**Comment by 2yangk23** [comment]
The Agility Plugin has things generic to all agility courses, but the stuff you added is specific to only one. You should try adding it in a way that is not intrusive to the existing plugin which will help to keep it maintainable if many things specific to individual courses are added. For example, you should try to move the regex/currentParts to within the Overlay class.

Also why does currentHeavyPart/currentLightPart need to be saved to a config? I never did this course before, but it seems like these values are always changing.

**Comment by Hyftar** [comment]
I think I did a good job of adding it in a way that is not intrusive to the AgilityPlugin code, I just reused a lot of the agility plugin code as a foundation for my add-on. Also I think the agility plugin has parts that doesn't apply to *all* courses such as mark of graces, so I don't think it is necessary to have it completely separated from the current agility plugin. I could indeed move the regex pattern to the overlay class, but I think it would just make the logic more obscure.

The values are kept in memory and will stay the same until you retrieve a new part, putting the values in a hidden config is a way for the client to keep track of the values through client restarts, therefore why it is in my opinion necessary to add those config variables.

**Pull request closed without merging**